csi/node_server.go

```diff
@@ -371,11 +371,21 @@ func (ns *NodeServer) nodePublishBlockVolume(volumeID, devicePath, targetPath st
 	if err := makeFile(targetPath); err != nil {
 		return status.Errorf(codes.Internal, "failed to create file %v: %v", targetPath, err)
 	}
+	defer func() {
+		if publishErr != nil {
+			if removeErr := os.Remove(targetPath); removeErr != nil {
+				log.WithError(removeErr).Errorf("failed to remove failed mount target %v: %v", targetPath, removeErr)
+			}
+		}
+	}()
 
 	log.Infof("Bind mounting device %v at %v", devicePath, targetPath)
 	if err := mounter.Mount(devicePath, targetPath, "", []string{"bind"}); err != nil {
-		if removeErr := os.Remove(targetPath); removeErr != nil {
-			return status.Errorf(codes.Internal, "failed to remove mount target %q: %v", targetPath, removeErr)
+		log.WithError(err).Errorf("failed to mount device %v to %v", devicePath, targetPath)
+		if isMounted, _ := mounter.IsMountPoint(targetPath); isMounted {
+			if unmountErr := mounter.Unmount(targetPath); unmountErr != nil {
+				log.WithError(unmountErr).Errorf("failed to unmount failed mount target %v", targetPath)
+			}
```

Comment:
Always try to remove the targetPath.

```suggestion
		log.WithError(err).Errorf("failed to mount device %v to %v", devicePath, targetPath)
		if isMounted, _ := mounter.IsMountPoint(targetPath); isMounted {
			if unmountErr := mounter.Unmount(targetPath); unmountErr != nil {
				log.WithError(unmountErr).Errorf("failed to unmount failed mount target %v", targetPath)
			}
			
		if removeErr := os.Remove(targetPath); removeErr != nil {
			return status.Errorf(codes.Internal, "failed to remove mount target %q: %v", targetPath, removeErr)
			...
		}
```

Reply:
The mount point removal for failed case is deferred before: https://github.com/longhorn/longhorn-manager/pull/4221/files#diff-e3208de2b41b1544cc85ea18e65fb55e847951129cdeee7db877af639c08e584R374-R380